core/lib/instrument.c

```diff
@@ -1955,6 +1955,14 @@ instrument_module_load_trigger(app_pc pc)
         if (ma != NULL && !TEST(MODULE_LOAD_EVENT, ma->flags)) {
             /* switch to write lock */
             os_get_module_info_unlock();
+
+            /* i#3385: re-initialize dynamic information, because it failed
+             * during the first flat-mmap that loaded the module.
+             */
+            if (!ma->os_data.have_dynamic_info) {
+                os_module_update_dynamic_info(ma->start, ma->end - ma->start, false);
+            }
```

Comment:
Would it be feasible to add a test for this?  By synthesizing a tiny ELF file with the right characteristics?  If you're not familiar with tooling to generate such a file, that's ok, we can punt on it: but please add a statement to the PR description explaining how you tested this, and maybe add a XXX comment here about possibly adding a regression test later.

Reply:
I'm not able to create a single tiny ELF file with late .dynstr section. I encountered this problem when testing oss-fuzz program: they applied `patchelf --set-rpath` command to shared libs. As a result their dynstr was relocated to the end.

I succeed to reproduce it on a small test-program by making it use a patched libc.so.6 library, but it weights 2.0M. I'm not familiar with DR test system, but to trigger bug a DR client is required that is perform some logic on loaded modules

Reply:
There are many test clients that register for module events (see `git grep _register_module_load suite`).  To avoid making a new one you could use `suite/tests/client-interface/events.dll.c`.

So `patchelf --set-rpath` didn't work on a tiny ELF file, only on the larger libc?  2.0M we could live with though.

For how to add the test: an example of a new app that uses an existing already-built events client is the fibers test:
```
  add_exe(client.fibers client-interface/fibers.c)
  torunonly_ci(client.fibers client.fibers client.events_cpp.dll
    client-interface/fibers.c "" "" "")
```

Reply:
Couldn't manage how to properly add a test. Provided a detailed description in this PR, attached compiled binaries and test sources, left a XXX comment in DR code.

Reply:
OK, though unfortunately it is likely no one will have resources later to add the test, we will go with this to get the fix in.